packages/react/react/src/hooks/useImageLazyLoading.ts

```diff
@@ -0,0 +1,70 @@
+import { useRefEffect } from './useRefEffect';
+
+interface Props {
+  src: string;
+  threshold?: number | number[];
+  root?: Document | Element | null;
+  rootMargin?: string;
+  onAction?: () => void;
+}
+
+/** @tossdocs-ignore */
+export function useImageLazyLoading<Element extends HTMLImageElement>({
```

Comment:
Thank you for the feedback. Upon further review, I realized that I am using the `useRefEffect` hook and returning the `ref` of that hook, so it would be appropriate to change the return type to `EffectRef<HTMLImageElement>`.